xbmc/video/VideoDatabase.cpp

```diff
@@ -3625,6 +3667,151 @@ void CVideoDatabase::SetMovieSet(int idMovie, int idSet)
     ExecuteQuery(PrepareSQL("update movie set idSet = null where idMovie = %i", idMovie));
 }
 
+void CVideoDatabase::GetMovieVersion(int idMovie, CFileItemList& items)
+{
+  if (m_pDB.get() == nullptr || m_pDS.get() == nullptr)
+    return;
+
+  try
+  {
+    m_pDS->query(PrepareSQL("SELECT type.name AS name,"
+                            "  path.strPath AS strPath,"
+                            "  files.strFileName AS strFileName "
+                            "FROM type"
+                            "  JOIN type_link ON"
+                            "    type_link.type_id = type.type_id"
+                            "  JOIN files ON"
+                            "    files.idFile = type_link.file_id"
+                            "  JOIN path ON"
+                            "    path.idPath = files.idPath "
+                            "WHERE type_link.media_id = %i", idMovie));
+
+    while (!m_pDS->eof())
+    {
+      CFileItemPtr pItem(new CFileItem());
+      pItem->SetLabel(m_pDS->fv(0).get_asString());
+      std::string file;
```

Comment:
I would declare `file` and `pItem` outside the `while` loop. No need to construct these objects on every iteration.

Reply:
Moved file out of the loop, the pItem is just a pointer and this code pattern used everywhere in the code, just keep the code style.